File: Minsk/ParenthesizedExpressionSyntax.cs
namespace Minsk;

public class ParenthesizedExpressionSyntax : ExpressionSyntax
{
    public SyntaxToken OpenParenthesisToken { get; }
    public ExpressionSyntax Expression { get; }
    public SyntaxToken CloseParenthesisToken { get; }

    public override SyntaxKind Kind => SyntaxKind.ParenthesizedExpression;

    public override IEnumerable<SyntaxNode> Children => new SyntaxNode[]
        { OpenParenthesisToken, Expression, CloseParenthesisToken };

    public ParenthesizedExpressionSyntax(SyntaxToken openParenthesisToken, ExpressionSyntax expression,
        SyntaxToken closeParenthesisToken)
    {
        OpenParenthesisToken = openParenthesisToken;
        Expression = expression;
        CloseParenthesisToken = closeParenthesisToken;
    }
}

File: kt/Minsk/src/main/kotlin/dev/phytolizer/minsk/ParenthesizedExpressionSyntax.kt
package dev.phytolizer.minsk

class ParenthesizedExpressionSyntax(
    val openParenthesisToken: SyntaxToken,
    val expression: ExpressionSyntax,
    val closeParenthesisToken: SyntaxToken
) : ExpressionSyntax() {
    override val kind: SyntaxKind
        get() = SyntaxKind.ParenthesizedExpression
    override val children: List<SyntaxNode>
        get() = listOf(openParenthesisToken, expression, closeParenthesisToken)
}
